google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -70,11 +74,36 @@
 
   private ArtifactPointer artifactPointer;
   private CommonModel commonModel;
+  private Sdk devAppServerJdk;
 
   private AppEngineModelSettings settings = new AppEngineModelSettings();
 
+  public AppEngineServerModel() {
+    initDefaultJdk();
+  }
+
+  @Nullable
+  private Sdk getCurrentProjectJdk() {
+    Project project = ProjectUtil.guessCurrentProject(null);
+    ProjectSdksModel projectJdksModel = new ProjectSdksModel();
+    projectJdksModel.reset(project);
+    return projectJdksModel.getProjectSdk();
+  }
+
+  private void initDefaultJdk() {
+    Sdk projectJdk = getCurrentProjectJdk();
+    if (projectJdk != null) {
+      setDevAppServerJdk(projectJdk);
+    }
+  }
+
   @Override
   public J2EEServerInstance createServerInstance() throws ExecutionException {
+    // FIXME: This keeps the dev_appserver's jdk in sync with the project jdk on behalf of the user.
```

Comment:
Sure - is that the recommended style for these sorts of TODOs? Might be nice if we could add that as a checkstyle rule.